march_hardware/include/march_hardware/PDOmap.h

```diff
@@ -16,13 +16,15 @@ namespace march
 struct IMCObject
 {
   uint16_t address;           // in IMC memory (see IMC manual)
-  uint16_t length;            // bits (see IMC manual)
+  uint8_t sub_index;          // sub index corresponding to PDO register (see IMC manual)
+  uint8_t length;             // bits (see IMC manual)
   uint32_t combined_address;  // combine the address(hex), sub-index(hex) and length(hex)
 
-  explicit IMCObject(uint16_t _address, uint16_t _length) : address(_address), length(_length)
+  explicit IMCObject(uint16_t _address, uint8_t _sub_index, uint8_t _length)
+    : address(_address), sub_index(_sub_index), length(_length)
   {
-    uint32_t MSword = ((address & 0xFFFF) << 16);  // Shift 16 bits left for most significant word
-    uint32_t LSword = (length & 0xFFFF);
+    uint32_t MSword = ((address & 0xFFFF) << 16);
```

Comment:
The bitwise and is actually not necessary, since it is already a 16 bit number.
```suggestion
    uint32_t MSword = (address << 16);
```